enhancements/network/ovn-observability-api.md

```diff
@@ -0,0 +1,262 @@
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=NetworkPolicy;AdminNetworkPolicy;EgressFirewall;UDNIsolation;MulticastIsolation
+type ObservabilityFeature string
+
+const (
+	NetworkPolicy      ObservabilityFeature = "NetworkPolicy"
+	AdminNetworkPolicy ObservabilityFeature = "AdminNetworkPolicy"
+	EgressFirewall     ObservabilityFeature = "EgressFirewall"
+	UDNIsolation       ObservabilityFeature = "UDNIsolation"
+	MulticastIsolation ObservabilityFeature = "MulticastIsolation"
+)
+```
+
+For now, we only have use cases for 2 different `ObservabilityConfig`s in a cluster, and ovn-kubernetes will likely have a
+limit for the number of CRs that will be handled.
+
+For the integration with netobserv, we could use a label on `ObservabilityConfig` to signal that configured samples should be
+reflected via netobserv. For example, `network.openshift.io/observability: "true"`.
+
+#### Observability filtering for debugging
+
+Work In Progress, will be included to the proposed CRD after consensus is reached.
+
+For debugging, we may need more granular filtering, like per-node or per-namespace filtering.
+
+- Per-node filtering may be implemented by adding a node selector to `ObservabilityConfig`, and is easy to implement in ovn-k
+as every node has its own ovn-k instance with OVN-IC.
+- Per-namespace filtering can only be applied to the namespaced Observability Features, like NetworkPolicy and EgressFirewall.
+Cluster-scoped features, like (Baseline)AdminNetworkPolicy, can only be additionally filtered based on name. Label selectors
+can't be used, as observability module doesn't watch any objects.
+
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+As long as netobserv can be installed on the management or hosted cluster, integration should work fine.
+If we need to make it work for shared-cluster netobserv deployment, more work.testing will be needed.
+Netobserv team confirmed that separate installations for management and hosted clusters are supported as a result of https://issues.redhat.com/browse/NETOBSERV-1196.
+
+For debuggability mode without netobserv integration, everything should work separately on every cluster with its own ovn-kubernetes instance.
+
+#### Standalone Clusters
+#### Single-node Deployments or MicroShift
+
+SNO should be supported with netobserv, Microshift is not.
+Microshift most likely will not be supported even without netobserv, as observability requires new features to be enabled
+and adds overhead to the cluster.
+
+### Implementation Details/Notes/Constraints
+
+Filtering for debugging use case may be tricky as explained in "Observability filtering for debugging" section.
+But it is important to implement, as OVN implementation works in a way, where performance optimizations can only be
+applied when there is only 1 `ObservabilityConfig` for a given feature. That means, adding a second `ObservabilityConfig`
```

Comment:
Unfortunately core OVN documentation doesn't mention the performance impact with multiple collectors attached to a single Sample/ACL but it should. I'll try to post a patch in ovn-org/ovn for that. If that lands in time it would be good to link to the core OVN documentation from here.